### PR TITLE
`proxy.istio.io/config` should attach in `annotations`

### PR DESCRIPTION
**Pull request by plucury**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [X] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by ericvn** [comment]
@istio/wg-policies-and-telemetry-maintainers Can you confirm?

**Comment by douglas-reid** [comment]
confirming that `proxy.istio.io/config` is an annotation ([documentation](https://istio.io/latest/docs/reference/config/annotations/)). this seems like a reasonable clarification.

File: content/en/docs/tasks/observability/distributed-tracing/mesh-and-proxy-config/index.md [hunk 1]
```diff
@@ -130,13 +130,15 @@ spec:
...
-                name: host
+      annotations:
+       ...
+       proxy.istio.io/config: |
+         tracing:
+           sampling: 10
+           custom_tags:
+             my_tag_header:
+               header:
+                 name: host
```
Comment (ericvn) [inline content/en/docs/tasks/observability/distributed-tracing/mesh-and-proxy-config/index.md:141]: I believe the indentation needs to be 2 more spaces?

Comment (ericvn) [reply content/en/docs/tasks/observability/distributed-tracing/mesh-and-proxy-config/index.md:141]: Thanks for the PR and we also need the CLA.

Comment (plucury) [reply content/en/docs/tasks/observability/distributed-tracing/mesh-and-proxy-config/index.md]: > I believe the indentation needs to be 2 more spaces?

fixed

**Comment by ericvn** [comment]
@plucury Thanks for the PR. If you can make sure the CLA is signed so that test passes, we can merge this in.

**Comment by plucury** [comment]
> Details

I've just signed the CLA. Is there any else I should do?

**Review by ericvn — APPROVED** [review]

**Pull request merged**